qualtran/resource_counting/classify_bloqs.py

```diff
@@ -109,7 +111,63 @@ def classify_t_count_by_bloq_type(
     return classified_bloqs
 
 
-def bloq_is_clifford(b: Bloq):
+_CLIFFORD_ANGLES = np.array(
+    [
+        np.pi,
+        np.pi / 2,
+        3 * np.pi / 2,
+        -np.pi,
+        -np.pi / 2,
+        -3 * np.pi / 2,
+        0.0,
+        -2 * np.pi,
+        2 * np.pi,
+    ]
+)
+_CLIFFORD_EXPONENTS = np.array([1.0, 0.5, 1.5, -1.0, -0.5, -1.5, 0.0, -2, 2])
+_T_ANGLES = np.array([np.pi / 4, -np.pi / 4])
+_T_EXPONENTS = np.array([0.25, -0.25])
+_ANGLE_ATOL = 1e-12
+
+
+def bloq_is_t_gate(b: Bloq) -> bool:
+    """Whether a bloq should be counted as a T gate.
+
+    This will return `True` for any instance of `TGate`. It will also consider
+    single-qubit rotations and return True if the angle corresponds to a T gate
+    (up to clifford reference frame).
+    """
+    from qualtran.bloqs.basic_gates import Rx, Ry, Rz, TGate, XPowGate, YPowGate, ZPowGate
+
+    if isinstance(b, TGate):
+        return True
+
+    if isinstance(b, (Rz, Rx, Ry)):
+        if is_symbolic(b.angle):
+            return False  # Symbolic rotation
+        if np.any(np.abs(b.angle - _T_ANGLES) < _ANGLE_ATOL):
+            return True  # T hidden in a rotation bloq
+        return False
```

Comment:
I think `bloq_is_t_gate` should return True only if the unitary of the bloq is equal to that of a `T` gate. For other single qubit rotations, we should have default decompositions; which a user can choose to override if needed; to avoid undercounting cliffords.

Reply:
we generally don't have "default" decompositions that one can override. 

I can change this function name to `bloq_is_t_like`. Counting Rx(pi/4) as 1 T gate is much more accurate than 1 "rotation", so I'm going to proceed with the change. We can add a decomposition of Rx to Rz later _or_ change the workflow where a re-write rule is responsible for simplifying any Rx(theta) to X/4 or whatever we like (and remove this itchy logic from the gate counting code).

Reply:
it's also unclear (to me) how we'd want to count cliffords for a situation like Rx(pi/4). Using an auto-correcting circuit like Fig 17 of game-of-surface-codes implies you can do any pauli with the same number of gates